### PR TITLE
Simplify Makefiles, make golangci-lint work again

### PR DESCRIPTION
**Pull request by kolyshkin**
This simplifies Makefiles, removing some old hacks, and makes golangci-linter work again. Please see individual commits for details.

Fixes: #1614

TODO (not in this PR):
 - re-add the disabled linters, going back to (at least) golangci-lint default set;
 - get rid of tests/tools entirely.

### CODE REVIEW

**Comment by kolyshkin** [comment]
OK, golangci-lint is now working: https://cirrus-ci.com/task/6600612519870464 (check "Run test").

**Comment by TomSweeneyRedHat** [comment]
@cevich PTAL

File: Makefile [hunk 1]
```diff
@@ -3,20 +3,31 @@ export GOPROXY=https://proxy.golang.org
...
 	local-test-integration \
 	local-test-unit \
 	local-validate \
-	lint \
-	vendor
+	test-integration \
+	test-unit \
+	validate \
+	vendor \
+	vendor-in-container
```
Comment (TomSweeneyRedHat) [inline Makefile:27]: I'm not sure we need to include vendor-in-container.

Comment (cevich) [reply Makefile:27]: Ya we do, it's a phony target - Meaning, it doesn't create a file called `vendor-in-container`.

**Comment by cevich** [comment]
Wow @kolyshkin this is really nice work and a very welcome set of improvements.  The only thing I'm slightly unsure of is the `GO111MODULES` change...

@TomSweeneyRedHat Do we need to maintain compatibility with (very) old golang due to backport possibility for some ancient (still supported) release?  I'm guessing not (they're probably pinned to specific commits), but wanted to double-check.

**Comment by TomSweeneyRedHat** [comment]
Thanks for the eyeballs @cevich 

As far as the old ancient Go goes, it's not outside of the realm of possibilities that we'd need to backport to an older version, but I'd say it's not very likely.  I wouldn't make older go a gating factor for this nor any other PR.

**Review by cevich — APPROVED** [review]
Thanks Tom.  K, LGTM then.

**Comment by kolyshkin** [comment]
> Wow @kolyshkin this is really nice work and a very welcome set of improvements. The only thing I'm slightly unsure of is the `GO111MODULES` change...

This is not needed since about go 1.14 (which is kinda ancient), and we've already made changes like this in other github.com/containers/* repos, e.g. https://github.com/containers/common/pull/993

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**